Update c# docs for release 0.11

jtattermusch opened the pull request:
-- change C# status to beta
-- change installation instructions on Linux (use debian packages).
-- make the readme.md style more similar to ruby's readme.
-- lots of nits

Missing:
route guide example for C# on grpc.io is out of date and needs to be updated.

tbetbetbe commented:
> route guide example for C# on grpc.io is out of date and needs to be updated.

What about the QuickStart (helloworld) in this repo? will that get updated when routeguide is updated?

jtattermusch commented:
Alright, I updated the helloworld README.md as well and fixed some of the links. PTAL.

In the next stage, I will create a PR to update contents of grpc.io site. Then I will delete the route guide's readme.md from this repo.
